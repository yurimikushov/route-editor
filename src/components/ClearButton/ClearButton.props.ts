type ClearButtonProps = {
  className?: string
  onClick: () => void
}

export default ClearButtonProps
