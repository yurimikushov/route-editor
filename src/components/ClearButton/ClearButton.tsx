import { FC } from 'react'
import cn from 'classnames'
import { ReactComponent as ClearIcon } from 'icons/clear.svg'
import ClearButtonProps from './ClearButton.props'

const ClearButton: FC<ClearButtonProps> = ({ className, onClick }) => {
  return (
    <button
      className={cn(
        className,
        'rounded-md text-gray-500 hover:text-black',
        'focus:outline-none focus-visible:ring-2 focus-visible:ring-indigo-500'
      )}
      type='reset'
      onClick={onClick}
    >
      <ClearIcon />
    </button>
  )
}

export default ClearButton
