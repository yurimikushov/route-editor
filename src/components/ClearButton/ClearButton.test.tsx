import renderer from 'react-test-renderer'
import noop from 'lodash/noop'
import ClearButton from './ClearButton'

it('should render clear button', () => {
  const tree = renderer.create(<ClearButton onClick={noop} />).toJSON()
  expect(tree).toMatchSnapshot()
})
