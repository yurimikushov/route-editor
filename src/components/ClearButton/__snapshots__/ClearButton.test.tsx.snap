// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render clear button 1`] = `
<button
  className="rounded-md text-gray-500 hover:text-black focus:outline-none focus-visible:ring-2 focus-visible:ring-indigo-500"
  onClick={[Function]}
  type="reset"
>
  <svg>
    clear.svg
  </svg>
</button>
`;
